src/Compilers/Core/Portable/CommandLine/ErrorLogger.cs

```diff
@@ -183,13 +170,30 @@ private Value GetLocationValue(Location location)
             return Value.Create(ImmutableArray.Create(wrapperKvp), this);
         }
 
+        private static string GetUri(SyntaxTree syntaxTree)
+        {
+            Uri uri;
+
+            if (!Uri.TryCreate(syntaxTree.FilePath, UriKind.RelativeOrAbsolute, out uri))
+            {
+                // The only constraint on SyntaxTree.FilePath is that it can be interpreted by
+                // various resolvers so there is no guarantee we can turn the arbitrary string
+                // in to a URI. If our attempt to do so fails, use the original string as the
+                // "URI".
+                return syntaxTree.FilePath;
```

Comment:
Above won't work directly because syntaxTree.FilePath and return value are strings, not Uris. I could introduce an extra string local and assign to in if/else or collapse to `return Uri.TryCreate(..) ? uri.ToString() : syntaxTree.FilePath`, but in my subjective view this is more readable than both of those, so I'll leave it as is.